src/ws/update-motd

```diff
@@ -1,24 +1,26 @@
-#!/bin/sh -e
+#!/bin/bash -e
 
-# syntax: update-motd [port [hostname [ipaddr [protocol]]]]
+# syntax: update-motd [port [hostname [protocol]]]
 # each argument can be given as the empty string to use the default
 
-# port number from cmdline, then systemctl file, then 9090
-# take the last Listen line; this will be the user-specified one
-port=${1:-$(systemctl show --property Listen cockpit.socket |
-              sed -E '$!d;$s/.*[^0-9]([0-9]+).*/\1/;')}
-port=${port:-9090}
-
 # hostname from cmdline, then `hostname -f`
 hostname=${2:-$(hostname -f || hostname)}
 
-# ip addr from cmdline, then default route source addr
-ip=${3:-$(ip -o route get 255.0 2>/dev/null | sed -e 's/.*src \([^ ]*\) .*/\1/')}
-
 # protocol from cmdline, then https
-protocol=${4:-https}
+protocol=${3:-https}
 
-hostname_url="${protocol}://${hostname}:${port}/"
-ip_url="${ip:+ or ${protocol}://${ip}:${port}/}"
+socket_path="\/*"
+empty_string=""
 
-printf 'Web console: %s%s\n\n' "${hostname_url}" "${ip_url}"  > /run/cockpit/active.motd
+printf 'Web console: \n' > /run/cockpit/active.motd
+# make newlines the only separator
+# IFS=$'\n'
+# port number from cmdline, then systemctl file
+for line in ${1:-$(systemctl show --value --property Listen cockpit.socket)}; do
+    port=${line##*:}
+    if [[ $port =~ [0-9]+ ]]; then
+        hostname_url="${protocol}://${hostname}:${port}/"
```

Comment:
The first scenario the user should deal with it. if they have such kind of configuration they might as well know where to connect. The main idea here to to provide information about the port in a nice way. And avoid showing IPs because it get messy.

The later I guess we can fix? Maybe somehow store the ports in a list and filter the duplicates out? I really dont know how to do this in bash, but I will take a look.